src/main/java/com/google/api/generator/engine/ast/GeneralForStatement.java

```diff
@@ -22,19 +22,11 @@
 
 @AutoValue
 public abstract class GeneralForStatement implements Statement {
-  public abstract Expr initializationExpr();
+  public abstract AssignmentExpr initializationExpr();
```

Comment:
Could you please confirm if this must be an AssignmentExpr (e.g. what about passing `foobar()` here)?

Reply:
Actually, it should not be limited to assignmentExpr, it should be expr which belongs to statmentExprList(https://docs.oracle.com/javase/specs/jls/se10/html/jls-14.html#jls).
My original though is to simplify the way to get variable. After consideration of back compatible, it should change back to expr. I will make change on this.

Reply:
@miraleung changed the interface use `Expr` for initialization. Before I merge the code, could you take one more look. Thanks.